crates/bevy_ecs/macros/src/component.rs

```diff
@@ -99,6 +99,11 @@ pub fn derive_buffered_event(input: TokenStream) -> TokenStream {
 }
 
 pub fn derive_resource(input: TokenStream) -> TokenStream {
+    // The resource derive *also* implements the Component trait
+    // We generate the Component implementation first, then add the Resource implementation,
+    // so then we can pick up all of the attributes from the Component derive
+    let component_derive_token_stream = derive_component(input.clone());
```

Comment:
I would really prefer not to derive `Component` for resources. Can't we just have a `struct ResourceComponent<R: Resource>(R);` instead?

Also, I'm not sure why the order matterrs here, it seems irrelevant.

Reply:
Much of the motivation for resources-as-components is being able to use tools like Observers and ComponentHooks for resources as well. For this, resources need to derive `Component`.